Kernel/Syscalls/socket.cpp

```diff
@@ -412,21 +412,23 @@ KResultOr<FlatPtr> Process::sys$socketpair(Userspace<const Syscall::SC_socketpai
         return result.error();
     auto pair = result.value();
 
-    int fds[2];
+    auto fd0_or_error = m_fds.allocate();
+    if (fd0_or_error.is_error())
+        return fd0_or_error.error();
     auto fd1_or_error = m_fds.allocate();
```

Comment:
Wouldn't that give us the same FD for both `allocate()` calls - considering that `allocate()` does not actually "allocate" the FD.

Reply:
Indeed. It seems to me that `allocate()` should return some sort of temporary holder of an allocated FD. This holder would release the allocation on destruction, but the FD could also be moved out of it.

Reply:
Good points, I cooked up a scoped type to handle the allocation / de-allocation and added tracking...